src/common/src/telemetry/report.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::Result;
+use risingwave_pb::meta::TelemetryInfoResponse;
+use tokio::sync::oneshot::Sender;
+use tokio::task::JoinHandle;
+use tokio::time::{interval, Duration};
+use uuid::Uuid;
+
+use super::{
+    post_telemetry_report, TelemetryReport, TELEMETRY_REPORT_INTERVAL, TELEMETRY_REPORT_URL,
+};
+
+#[async_trait::async_trait]
+pub trait TelemetryInfoFetcher {
+    async fn fetch_telemetry_info(&self) -> Result<TelemetryInfoResponse>;
+}
+
+pub trait TelemetryReportCreator {
+    // inject dependencies to impl structs if more metrics needed
+    fn create_report(
+        &self,
+        tracking_id: String,
+        session_id: String,
+        up_time: u64,
+    ) -> Result<impl TelemetryReport>;
+
+    fn report_type(&self) -> &str;
+}
+
+pub fn start_telemetry_reporting<F, I>(
```

Comment:
So every node is looping to proactivaly fetch telemetry from meta service. This looks like an uncessary RPC burden for the meta service. Is it possible to issue the "start_telemetry_reporting" request from Meta service?